src/Umbraco.Web.UI.Client/src/views/propertyeditors/blocklist/blocklist.component.js

```diff
@@ -0,0 +1,536 @@
+                        vm.setBlockFocus(vm.layout[Math.max(createIndex-1, 0)].$block);
+                    }
+
+                    editorService.close();
+                }
+            };
+
+            blockPickerModel.clickClearClipboard = function ($event) {
+                clipboardService.clearEntriesOfType("elementType", vm.availableContentTypesAliases);
+                clipboardService.clearEntriesOfType("elementTypeArray", vm.availableContentTypesAliases);
+            };
+
+            blockPickerModel.clipboardItems = [];
+
+            var singleEntriesForPaste = clipboardService.retriveEntriesOfType("elementType", vm.availableContentTypesAliases);
+            singleEntriesForPaste.forEach(function (entry) {
+                blockPickerModel.clipboardItems.push(
+                    {
+                        type: "elementType",
+                        pasteData: entry.data,
+                        blockConfigModel: modelObject.getScaffoldFromAlias(entry.alias),
+                        elementTypeModel: {
+                            name: entry.label,
+                            icon: entry.icon
+                        }
+                    }
+                );
+            });
+            
+            var arrayEntriesForPaste = clipboardService.retriveEntriesOfType("elementTypeArray", vm.availableContentTypesAliases);
+            arrayEntriesForPaste.forEach(function (entry) {
+                blockPickerModel.clipboardItems.push(
+                    {
+                        type: "elementTypeArray",
+                        pasteData: entry.data,
+                        blockConfigModel: {}, // no block configuration for paste items of elementTypeArray.
+                        elementTypeModel: {
+                            name: entry.label,
+                            icon: entry.icon
+                        }
+                    }
+                );
+            });
+
+            // open block picker overlay
+            editorService.open(blockPickerModel);
+
+        };
+
+        function requestCopyBlock(block) {
+            clipboardService.copy("elementTypeArray", block.content.contentTypeAlias, block.content, block.label);
+        }
+
+        var requestCopyAllBlocks = function() {
+
+            var elementTypesToCopy = vm.layout.filter(entry => entry.$block.config.unsupported !== true).map(entry => entry.$block.content);
+            
+            // list aliases
+            var aliases = elementTypesToCopy.map(content => content.contentTypeAlias);
+
+            // remove dublicates
+            aliases = aliases.filter((item, index) => aliases.indexOf(item) === index);
+            
+            var contentNodeName = "";
+            if(vm.umbVariantContent) {
+                contentNodeName = vm.umbVariantContent.editor.content.name;
+            }
+            // TODO: check if we are in an overlay and then lets get the Label of this block.
```

Comment:
Is this TODO still relevant?

Reply:
Yes, we need to gather a useful name for the Clipboard entry, when copying the whole property we use the name of the content-node, but when the block-editor is within another block editor I thought it would make sense to gather the label from the block that it's contained within.
Also because we don't have the umbVariantContent requirement in this scenario, at least not when editing in infinite editors.
Should we get that removed and create a task on it?

Reply:
@nielslyngsoe I'm not really sure, that is what I'm asking you here. I see a new TODO in the code which normally indicates that something isn't done and needs to be completed. So if this is something that needs to be completed for the block editor to work then yes we need to get it done, otherwise if this is a TODO for the future then perhaps a task is needed, but that's your call to make.

Reply:
I have just fixed it now and thereby removed this TODO.